src/category_theory/monoidal/rigid.lean

```diff
@@ -0,0 +1,275 @@
+/-
+Copyright (c) 2021 Jakob von Raumer. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jakob von Raumer
+-/
+
+import category_theory.monoidal.category
+
+
+/-!
+# Rigid (autonomous) monoidal categories
+
+This file defines rigid (autonomous) monoidal categories and the necessary theory about
+exact pairings and duals.
+
+## Main definitions
+
+* `exact_pairing` of two objects of a monoidal category
+* Type classes `has_left_dual` and `has_right_dual` that capture that a pairing exists
+* The `right_adjoint_mate f` as a morphism `f^* : Y^* ⟶ X^*` for a morphism `f : X ⟶ Y`
+* The classes of `right_rigid_category`, `left_rigid_category` and `rigid_category`
+
+## Main statements
+
+* `comp_right_adjoint_mate`: The adjoint mates of the composition is the composition of
+  adjoint mates.
+
+## Notations
+
+* `η_` and `ε_` denote the coevaluation and evaluation morphism of an exact pairing.
+* `X^*` and `*^X` denote the right and left dual of an object, as well as the adjoint
+  mate of a morphism.
+
+## Future work
+
+* Show that `X ⊗ Y` and `Y^* ⊗ X^*` form an exact pairing.
+* Show that the left adjoint mate of the right adjoint mate of a morphism is the morphism itself.
+* Simplify constructions in the case where a symmetry or braiding is present.
+
+## References
+
+* <https://ncatlab.org/nlab/show/rigid+monoidal+category>
+
+## Tags
+
+rigid category, monoidal category
+
+-/
+open category_theory
+
+universes v v₁ v₂ v₃ u u₁ u₂ u₃
+noncomputable theory
+
+namespace category_theory
+
+variables {C : Type u₁} [category.{v₁} C] [monoidal_category C]
+
+/-- An exact pairing is a pair of objects `X Y : C` which admit
+  a coevaluation and evaluation morphism which fulfill two triangle equalities. -/
+class exact_pairing (X Y : C) :=
+  (coevaluation [] : 𝟙_ C ⟶ X ⊗ Y)
+  (evaluation [] : Y ⊗ X ⟶ 𝟙_ C)
+  (coevaluation_evaluation' [] :
+    (𝟙 Y ⊗ coevaluation) ≫ (α_ _ _ _).inv ≫ (evaluation ⊗ 𝟙 Y)
+    = (ρ_ Y).hom ≫ (λ_ Y).inv . obviously)
+  (evaluation_coevaluation' [] :
+    (coevaluation ⊗ 𝟙 X) ≫ (α_ _ _ _).hom ≫ (𝟙 X ⊗ evaluation)
+    = (λ_ X).hom ≫ (ρ_ X).inv . obviously)
```

Comment:
Usually we wouldn't indent here.

Reply:
Are you sure? It looks this way in `monoidal_category`.

Reply:
It seems the style guide says "In a class or structure definition, we do not indent fields, as in: ...", but at the same time you are absolutely right that I broke this when writing `monoidal_category`. :-)

Reply:
Ah I thought you just meant the line breaks within the fields!